### PR TITLE
fix(middleware): fix a bug which middlewares array will reverse more …

### PR DESCRIPTION
**Pull request by marshalYuan**
if reverse middlewares array every time,then
```
const middleware1 = store => next => action => {console.log(1);return next(action)}
const middleware2 = store => next => action => {console.log(2);return next(action)}
const middleware3 = store => next => action => {console.log(3);return next(action)}

const store = createStore({count:1}, applyMiddlewar(middleware1, middleware2, middleware3))

actions.increment() // 1,2,3
actions.increment() // 3,2,1
actions.increment() // 1,2,3
```

### CODE REVIEW

File: src/middleware/applyMiddleware.ts [hunk 1]
```diff
@@ -5,13 +5,13 @@ const finalMiddleware = (store: Store, args) => (action: Function) =>
   set(store, action(store.getState(), ...args))
 
 export default function applyMiddleware(...middlewares) {
+  middlewares = middlewares.reverse() // just reverse once
```
Comment (matheusml) [inline src/middleware/applyMiddleware.ts]: @marshalYuan could you please remove the comment?

**Comment by matheusml** [comment]
Thanks for this @marshalYuan 

Could you please:
- Add this change to the CHANGELOG
- Bump the npm version to 4.8.1

File: src/middleware/applyMiddleware.ts (hunk 1 above)
Comment (marshalYuan) [reply src/middleware/applyMiddleware.ts]: 👌

**Comment by matheusml** [comment]
Thanks @marshalYuan 

💯

**Pull request merged**